src/sdk/cart/validate.ts

```diff
@@ -58,15 +58,42 @@ export const ValidateCartMutation = gql`
         name
       }
       gtin
+      additionalProperty {
+        propertyID
+        name
+        value
+        valueReference
+      }
     }
   }
 `
 
 export const isGift = (item: CartItem) => item.price === 0
 
+const serializeAdditionalProperty = (itemOffered: CartItem['itemOffered']) => {
+  if (
+    !itemOffered.additionalProperty ||
+    itemOffered.additionalProperty?.length === 0
+  ) {
+    return null
+  }
+
+  return itemOffered.additionalProperty
+    .map(({ propertyID }) => propertyID)
+    .join('-')
+}
+
 export const getItemId = (
   item: Pick<CartItem, 'itemOffered' | 'seller' | 'price'>
-) => `${item.itemOffered.sku}:${item.seller.identifier}:${item.price}`
+) =>
+  [
+    item.itemOffered.sku,
+    item.seller.identifier,
+    item.price,
+    serializeAdditionalProperty(item.itemOffered),
+  ]
+    .filter(Boolean)
```

Comment:
do we need to `filter`? If addicionalProperty is null, than the final key will be `...::null` which doesn't impact the final key. What do you think?